modules/xcm-bridge-hub-router/src/lib.rs

```diff
@@ -100,67 +103,111 @@ pub mod pallet {
 	#[pallet::hooks]
 	impl<T: Config<I>, I: 'static> Hooks<BlockNumberFor<T>> for Pallet<T, I> {
 		fn on_initialize(_n: BlockNumberFor<T>) -> Weight {
-			// if XCM queue is still congested, we don't change anything
-			if T::WithBridgeHubChannel::is_congested() {
-				return T::WeightInfo::on_initialize_when_congested()
-			}
+			Bridge::<T, I>::mutate(|bridge| {
+				// TODO: make sure that `WithBridgeHubChannel::is_congested` returns true if either
+				// of XCM channels (outbound/inbound) is suspended. Because if outbound is suspended
+				// that is definitely congestion. If inbound is suspended, then we are not able to
+				// receive the "report_bridge_status" signal (that maybe sent by the bridge hub).
+
+				// if the channel with sibling/child bridge hub is suspended, we don't change
+				// anything
+				if T::WithBridgeHubChannel::is_congested() {
+					return T::WeightInfo::on_initialize_when_congested()
+				}
 
-			DeliveryFeeFactor::<T, I>::mutate(|f| {
-				let previous_factor = *f;
-				*f = InitialFactor::get().max(*f / EXPONENTIAL_FEE_BASE);
-				if previous_factor != *f {
-					log::info!(
-						target: LOG_TARGET,
-						"Bridge queue is uncongested. Decreased fee factor from {} to {}",
-						previous_factor,
-						f,
-					);
+				// if bridge has reported congestion, we don't change anything
+				if bridge.is_congested {
+					return T::WeightInfo::on_initialize_when_congested()
+				}
 
-					T::WeightInfo::on_initialize_when_non_congested()
-				} else {
-					// we have not actually updated the `DeliveryFeeFactor`, so we may deduct
-					// single db write from maximal weight
-					T::WeightInfo::on_initialize_when_non_congested()
-						.saturating_sub(T::DbWeight::get().writes(1))
+				// if fee factor is already minimal, we don't change anything
+				if bridge.delivery_fee_factor == FixedU128::one() {
```

Comment:
I see we use `FixedU128::one()` on several places and also for default vaule: `BridgeState { delivery_fee_factor: FixedU128::one()`
is it worth to introduce some constant like MINIMAL_DELIVERY_FEE_FACTOR and use that instead of `FixedU128::one()` everywhere?

Reply:
done